pallets/gear/src/lib.rs

```diff
@@ -774,7 +788,7 @@ pub mod pallet {
             T::Currency::reserve(&who, value.unique_saturated_into())
                 .map_err(|_| Error::<T>::NotEnoughBalanceForReserve)?;
 
-            if common::program_exists(destination) {
+            if common::program_exists(destination) | common::paused_program_exists(destination) {
```

Comment:
Good remark, thanks! I check how to refactor it.

Reply:
I thought about it a bit and it looks like current solution with different storage key is better. To store under the same key an enumeration has to be introduced, however we already have `Program`, `ActiveProgram`, `TerminatedProgram` and so on. However if you/team insists to refactor this way - I'll change it.

Another point is that even if `PausedProgram` would be merged into the same storage key we still could need such function to determine if the program paused/running.

At the moment I could suggest to call `paused_program_exists` by `program_exists`. This will eliminate such conditions and fits the current business logic.

What should I do?